Assets/Scripts/SS3D/Systems/Chat/ChatMessageSender.cs

```diff
@@ -0,0 +1,146 @@
+﻿using FishNet;
+using FishNet.Connection;
+using JetBrains.Annotations;
+using SS3D.Core;
+using SS3D.Logging;
+using SS3D.Permissions;
+using SS3D.Systems.Entities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace SS3D.Engine.Chat
+{
+    public static class ChatMessageSender
+    {
+        private const string ChatLogFolderName = "Chat";
+        
+        private static readonly string ChatLogPath = $"{UnityEngine.Application.dataPath}/../Logs/{ChatLogFolderName}.txt";
+        private static readonly List<ChatWindow> ChatWindows = new List<ChatWindow>();
+        
+        private static bool Initialized;
+
+        public static void InitializeIfNeeded()
+        {
+            if (Initialized)
+            {
+                return;
+            }
+            
+            InstanceFinder.ClientManager.RegisterBroadcast<ChatMessage>(OnClientReceiveChatMessage);
+            InstanceFinder.ServerManager.RegisterBroadcast<ChatMessage>(OnServerReceiveChatMessage);
+            
+            Initialized = true;
+        }
+
+        public static void RegisterChatWindow(ChatWindow chatWindow)
```

Comment:
Why does this class handles UI? Sending chat messages to chat channels is purely data-related, UI should be handled elsewhere to avoid multiple responsibility on this class.